bindings/zeebe/command/create_instance.go

```diff
@@ -75,9 +80,19 @@ func (z *ZeebeCommand) createInstance(ctx context.Context, req *bindings.InvokeR
 		}
 	}
 
-	response, err := cmd3.Send(ctx)
+	var response interface{}
+	if payload.WithResult && payload.RequestTimeout.Duration != time.Duration(0) {
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, payload.RequestTimeout.Duration)
+		defer cancel()
+		response, err = cmd3.WithResult().FetchVariables(payload.FetchVariables...).Send(ctxWithTimeout)
+	} else if payload.WithResult {
+		response, err = cmd3.WithResult().FetchVariables(payload.FetchVariables...).Send(ctx)
+	} else {
+		response, err = cmd3.Send(ctx)
+	}
+
```

Comment:
The timeout has only an affect if withResult is used. Using `withResult` means that the operation is synchronous instead of asynchronous. And the request timeout defines how long the client should wait for the workflow/process to finish to get the result.

From a code perspective, there are two ´send` methods in the client. One if `withResult` was used and which gets the timeout from the context and another one which will not use any timeout.

# without request timeout
https://github.com/camunda/zeebe/blob/71d2e6accc9f0452c72c1de7c16f18cc24d22552/clients/go/pkg/commands/create_instance.go#L165

# with request timeout
https://github.com/camunda/zeebe/blob/71d2e6accc9f0452c72c1de7c16f18cc24d22552/clients/go/pkg/commands/create_instance.go#L174

Reply:
I can add a comment to make this more clear

Reply:
Thanks for explaining